streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -186,7 +206,13 @@ public LRUCacheEntry delete(final String namespace, final Bytes key) {
             return null;
         }
 
-        return cache.delete(key);
+        final LRUCacheEntry entry;
+        synchronized (cache) {
+            final long oldSize = cache.sizeInBytes();
+            entry = cache.delete(key);
```

Comment:
in this case we can't the change in bytes, but because `delete` has no danger of side effects affecting the cache size (eg as in `put` which may trigger an eviction, so the delta isn't just the bytes passed in to `put`) we can just compute the change in size directly by subtracting the size of this, which should be known, right?